src/test/regress/sql/reassure_owned.sql

```diff
@@ -0,0 +1,70 @@
+CREATE ROLE role1;
+CREATE ROLE role2;
+
+GRANT CREATE ON SCHEMA public TO role1;
+
+SET ROLE role1;
+CREATE TABLE public.test_table (col1 int);
+RESET ROLE;
+select create_distributed_table('test_table', 'col1');
+
+SELECT result from run_command_on_all_nodes(
+  $$
+  SELECT jsonb_agg(to_jsonb(q2.*)) FROM (
+   SELECT
+    schemaname,
+    tablename,
+    tableowner
+FROM
+    pg_tables
+WHERE
+    tablename = 'test_table'
+  ) q2
+  $$
+) ORDER BY result;
+
+set citus.log_remote_commands to on;
+set citus.grep_remote_commands = '%REASSIGN OWNED BY%';
+REASSIGN OWNED BY role1 TO role2;
```

Comment:
1. Regardless of what we decide to do with the scenarios C1,C2,C3,C4,C5,C6, 
we should have at least one test for each scenario. And also we should ensure to have test cases with >1 role in the BY clause, e.g. 2 roles: `REASSIGN OWNED BY role0, role1 TO role2;`

2. As always, let's also add tests where the role names have escaped characters :)